src/curl.c

```diff
@@ -116,6 +139,11 @@ static int check_connection(const char *test_capath)
 		return -1;
 	}
 
+	curl_ret = swupd_curl_set_basic_options(curl, content_url);
```

Comment:
Use version_url here so you can remove the line 121 (set CURLOPT_URL).
You can remove swupd_curl_set_optional_client_cert() call too, because this is also set by swupd_curl_set_basic_options()

Reply:
you are right, actually there were another couple more options that were already set in swupd_curl_set_basic_options(), I removed those too